roles/repo_setup/tasks/ci_mirror.yml

```diff
@@ -12,6 +12,8 @@
     set -o pipefail
     source /etc/ci/mirror_info.sh
     sed -i -e "s|https://trunk.rdoproject.org|$NODEPOOL_RDO_PROXY|g" *.repo
+    {% if ansible_distribution | lower == 'centos' and ansible_distribution_major_version is not version('10', '==') %}
```

Comment:
Can you add additional info about location of CentOS10 nodepool? If it doesn't exist, please create another `when` condition. Instead of nesting `if/else` here,  please modify the line 10, and extend it over there. For the Centos10 specific use-case, please add another task.
You might consider using nested conditions, something like:
```
- block
   - name: cs9
     when: major_version is < 10
   - name: cs10
     when: major_version is 10
   when:
     - mirror_path.stat.exists
```